### PR TITLE
Added Gherkin tests and Exploratory notes

### PR DESCRIPTION
**Pull request by adrielgamorim**

### CODE REVIEW

File: APITestingFramework/Features/Projects/Post.feature [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+        When the user sends a POST request to the "/projects" endpoint
+        Then the response should have a status code of 201
+        And the response should contain the details of the newly created project
+
+    Scenario: Attempt to create a new project without valid parameters
+        Given the user is authenticated
+        And the request body does not contain the parameter "Content"
+        When the user sends a POST request to the "/projects" endpoint
+        Then the response should have a status code of 400
+        And the response should contain an error message indicating the parameter "Content" is required
```
Comment (Anaeli) [inline APITestingFramework/Features/Projects/Post.feature]: use space  before

File: APITestingFramework/Features/Project/GetAll.feature [hunk 2]
```diff
@@ -0,0 +1,13 @@
+Feature: Retrieve all projects
+
```
Comment (Anaeli) [inline APITestingFramework/Features/Project/GetAll.feature:2]: Precondition:

before_feature
user created

in feature:
 1. Post a list of projects

File: APITestingFramework/Features/Project/GetById.feature [hunk 3]
```diff
@@ -0,0 +1,21 @@
+Feature: Retrieve a project by id
+
```
Comment (Anaeli) [inline APITestingFramework/Features/Project/GetById.feature:2]: tag your scenarios

File: APITestingFramework/Features/Project/GetById.feature [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    Scenario: Retrieve a specific project with valid ID and valid authentication
+        Given the user is authenticated
+        And the project with ID "1" exists
+        When the user sends a GET request to the "/projects/1" endpoint
+        Then the response should have a status code of 200
+        And the response should contain the details of the project with ID "1"
```
Comment (Anaeli) [inline APITestingFramework/Features/Project/GetById.feature:8]: which details?

File: APITestingFramework/Features/Project/GetById.feature [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+    Scenario: Attempt to retrieve a project with invalid ID
+        Given the user is authenticated
+        When the user sends a GET request to the "/projects/9999" endpoint
+        Then the response should have a status code of 404
+        And the response should contain an error message indicating the project was not found
```
Comment (Anaeli) [inline APITestingFramework/Features/Project/GetById.feature:14]: the response should contain an error message indicating "the project was not found"
Use "..." to highlight the error message to validate

File: APITestingFramework/Features/Project/Post.feature [hunk 6]
```diff
@@ -0,0 +1,22 @@
+Feature: Create a project
+
+    Scenario: Create a new project with valid parameters and valid authentication
+        Given the user is authenticated
+        And the request body contains the parameter "Content" with value "Project 1"
+        When the user sends a POST request to the "/projects" endpoint
+        Then the response should have a status code of 201
+        And the response should contain the details of the newly created project
```
Comment (Anaeli) [inline APITestingFramework/Features/Project/Post.feature:8]: specify the details to validate

File: APITestingFramework/Features/Project/Post.feature [hunk 7]
```diff
@@ -0,0 +1,22 @@
+Feature: Create a project
+
+    Scenario: Create a new project with valid parameters and valid authentication
+        Given the user is authenticated
+        And the request body contains the parameter "Content" with value "Project 1"
+        When the user sends a POST request to the "/projects" endpoint
```
Comment (Anaeli) [inline APITestingFramework/Features/Project/Post.feature:6]: And the user fill the body with following data:
"""
{ 
  name: test
}

Comment (Anaeli) [reply APITestingFramework/Features/Project/Post.feature:6]: in this part you should log and keep the project name and ID

File: APITestingFramework/Features/Projects/Post.feature (hunk 1 above)
Comment (adrielgamorim) [reply APITestingFramework/Features/Projects/Post.feature]: Done in next commit

File: APITestingFramework/Features/Project/GetById.feature (hunk 5 above)
Comment (adrielgamorim) [reply APITestingFramework/Features/Project/GetById.feature:14]: Done in next commit

File: APITestingFramework/Features/Project/Post.feature (hunk 6 above)
Comment (adrielgamorim) [reply APITestingFramework/Features/Project/Post.feature:8]: Done in next commit

File: APITestingFramework/Features/Project/GetById.feature (hunk 4 above)
Comment (adrielgamorim) [reply APITestingFramework/Features/Project/GetById.feature:8]: Done in next commit

File: APITestingFramework/Features/Project/Post.feature (hunk 7 above)
Comment (adrielgamorim) [reply APITestingFramework/Features/Project/Post.feature:6]: Done in the next commit

File: APITestingFramework/Features/Project/GetById.feature (hunk 3 above)
Comment (adrielgamorim) [reply APITestingFramework/Features/Project/GetById.feature:2]: Done in next commit

File: APITestingFramework/Features/Project/GetAll.feature (hunk 2 above)
Comment (adrielgamorim) [reply APITestingFramework/Features/Project/GetAll.feature:2]: I couldn't properly understand Eliana, could you please elaborate further?

Comment (Anaeli) [reply APITestingFramework/Features/Project/GetAll.feature:2]: you should POST at least 2 projects to https://todo.ly/api/projects.json enpoint
and keep the id from this 2 projects
i.e.
Content = Project1
Content = Project 2

File: APITestingFramework/Features/Project/GetAll.feature [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+    Scenario: Retrieve all projects with valid authentication
+        Given the user is authenticated
+        When the user sends a GET request to the "/projects" endpoint
+        Then the response should have a status code of 200
+        And the response should contain a list of all projects
```
Comment (Anaeli) [inline APITestingFramework/Features/Project/GetAll.feature:7]: And the response should contains "2" elements
And the response have the following projects:
  | Project Name|
   |project 1       |
   |project 2       |

**Pull request closed without merging**